todo_app/app.py

```diff
@@ -1,44 +1,85 @@
 from flask import Flask, render_template, url_for, redirect, request
-from .data.session_items import get_items, add_item
+from flask_login.utils import login_user
+from werkzeug.datastructures import Accept
+
+from todo_app.data.user import User
 import pymongo
 from datetime import datetime
 from bson.objectid import ObjectId
+from flask_login import LoginManager, login_required, current_user
+from oauthlib.oauth2 import WebApplicationClient
 
-
-#from todo_app.flask_config import Config
+from todo_app.flask_config import Config
 
 
 import requests
+import os
+import functools
+
 
 from todo_app.ToDoCard import ToDoCard
 from todo_app.ViewModel import ViewModel
 
 import os
 
+
+def writer_required(func):
+    @functools.wraps(func)
+    def writer_check():
+        if current_user.user_id != os.getenv("USER_ID"):
```

Comment:
You haven't actually set a user_id on your User class - if you look at the class you'll see you have id or isrole - we can use either here e.g.
```python
        if current_user.isrole != "writer":
```
though making this run will break your end-to-end tests! One way to resolve that might be:
```suggestion
        if os.getenv('LOGIN_DISABLED') != 'True' and current_user.isrole != "writer":
```

Reply:
Thought it came as part of `_init_`, still I can make the changes